lib/full-site-editing/block-templates.php

```diff
@@ -360,7 +382,21 @@ function gutenberg_get_block_templates( $query = array(), $template_type = 'wp_t
 		}
 	}
 
-	return $query_result;
+	/**
+	 * Filters the array of queried block templates array after they've been fetched.
+	 *
+	 * @since 5.8
+	 *
+	 * @param WP_Block_Template[] $query_result Array of found block templates.
+	 * @param array $query {
+	 *     Optional. Arguments to retrieve templates.
+	 *
+	 *     @type array  $slug__in List of slugs to include.
+	 *     @type int    $wp_id Post ID of customized template.
+	 * }
+	 * @param array $template_type wp_template or wp_template_part.
+	 */
+	return apply_filters( 'queried_block_templates', $query_result, $query, $template_type );
```

Comment:
I think the name here should correspond to the same name as the `pre_*` filter above.  So we would have `pre_get_block_templates` and `get_block_templates` (similar to as you did in the functions below).

Also, im not completely certain, but I wonder if we should be adding the `gutenberg` prefix to these filters the same as they are in the functions they are contained in?  Like `pre_gutenberg_get_block_templates` / `gutenberg_get_block_templates` etc?  It would be a little more specific to their use and hopefully less potential of conflict 🤔

Reply:
As I said, I'm also a bit unsure about the names of these filter names, and this one the most.

Searching the project for filters with a [gutenberg_](https://github.com/WordPress/gutenberg/blob/94395faadb8be74fbcf7791abac586cf7dc54c71/lib/widgets.php#L41) prefix I only found 1. Blocks templates themself are pretty new territory, so I doubt there would be conflicts.

I'll change it to `get_block_templates` for at least now, if anybody has a better suggestion (for any of the filters) I'll change it again.